packages/shared/src/mock/utils.ts

```diff
@@ -61,14 +61,15 @@ export function findMockEntity(path: string): ProvidedEntity | undefined {
   return child;
 }
 
+// eslint-disable-next-line etc/no-misused-generics
```

Comment:
Here, I'm explicitly allowing the caller to type-cast via the generic, as this function is used in Storybook for convenience and with known data. Note that it still works without `<T>` below because of `as T[]` on the last line.